reconcile/utils/pagerduty_api.py

```diff
@@ -1,40 +1,90 @@
-import datetime
+import logging
+from collections.abc import Iterable
+from datetime import datetime as dt
+from typing import Callable
+from typing import Optional
+from typing import Protocol
```

Comment:
Protocol and Callable should be imported from `collections.abc` too, poor things 😬

Reply:
`Callable` yes, `Protocol` no. Why do we have `collections.abc` and `typings`!?

Reply:
yeah, you're right

The answer for that is in https://peps.python.org/pep-0585/